src/Microsoft.Health.Fhir.Shared.Api/Features/Formatters/FormatParametersValidator.cs

```diff
@@ -147,5 +145,36 @@ private async Task<bool> IsFormatSupportedAsync(ResourceFormat resourceFormat)
                 }
             });
         }
+
+        public void CheckPrettyParameter(HttpContext httpContext)
+        {
+            var prettyParameterValue = GetParameterValueFromQueryString(httpContext, KnownQueryParameterNames.Pretty);
+
+            if (prettyParameterValue != null && !bool.TryParse(prettyParameterValue, out _))
+            {
+                throw new BadRequestException(Api.Resources.InvalidPrettyParameter);
+            }
+        }
+
+        public void CheckSummaryParameter(HttpContext httpContext)
+        {
+            var summaryParameterValue = GetParameterValueFromQueryString(httpContext, KnownQueryParameterNames.Summary);
+
+            if (summaryParameterValue != null && !Enum.TryParse<SummaryType>(summaryParameterValue, true, out _))
+            {
+                throw new BadRequestException(Api.Resources.InvalidSummaryParameter);
```

Comment:
For summary the new error message contains less information than the old one:
New: "diagnostics": "The requested \"_summary\" parameter is invalid."
Old: "diagnostics": "Invalid _summary: 'badValue' is not a recognized summary value"

Should we include the value that couldn't be parsed?

Reply:
I also don't like that we 'support' summary but it doesn't seem to do anything for values other than count (or at least not what the spec says it should do)

Reply:
Let me update to include value, and maybe allowed values as well.

---
In reply to: [587736868](https://github.com/microsoft/fhir-server/pull/1699#discussion_r587736868) [](ancestors = 587736868)

Reply:
Oy vey, our output formatters ignore summary whatsoever. Let me tinker with that

---
In reply to: [587738920](https://github.com/microsoft/fhir-server/pull/1699#discussion_r587738920) [](ancestors = 587738920)